### PR TITLE
Fix subscription throughput data generation [11964]

### PR DESCRIPTION
**Pull request by jparisu**

### CODE REVIEW

File: src/cpp/rtps/reader/StatelessReader.cpp [hunk 1]
```diff
@@ -238,12 +238,15 @@ bool StatelessReader::matched_writer_is_matched(
 bool StatelessReader::change_received(
         CacheChange_t* change)
 {
-    // Only make visible the change if there is not other with bigger sequence number.
+    // std::cout << "[FAST STATELESS READER] change_received" << std::endl;
```
Comment (MiguelCompany) [inline src/cpp/rtps/reader/StatelessReader.cpp:241]: Seems this was left here unintentionally

File: src/cpp/rtps/reader/StatelessReader.cpp [hunk 2]
```diff
@@ -252,13 +255,15 @@ bool StatelessReader::change_received(
...
             {
+                // WARNING! This method could destroy the change
                 getListener()->onNewCacheChangeAdded(this, change);
             }
 
             new_notification_cv_.notify_all();
 
             // statistics callback
-            on_subscribe_throughput(change->serializedPayload.length);
+            // std::cout << "[FAST STATELESS READER] on_subscribe_throughput sending thr" << std::endl;
```
Comment (MiguelCompany) [inline src/cpp/rtps/reader/StatelessReader.cpp:265]: Seems this was left here unintentionally

File: test/blackbox/common/DDSBlackboxTestsStatistics.cpp [hunk 3]
```diff
@@ -166,9 +156,52 @@ TEST(DDSStatistics, simple_statistics_datareaders)
...
+    // Create parameters to iterate over every Statistics kind
+    // The test is separated between the statistics retrieved by a DataWriter or a DataReader
+    std::vector<std::tuple<std::string, std::string, std::size_t>> writer_statistics_kinds = {
+        {"DATA_COUNT_TOPIC",                statistics::DATA_COUNT_TOPIC,               num_samples},
+        {"RTPS_SENT_TOPIC",                 statistics::RTPS_SENT_TOPIC,                num_samples},
+        {"NETWORK_LATENCY_TOPIC",           statistics::NETWORK_LATENCY_TOPIC,          num_samples},
+        {"PUBLICATION_THROUGHPUT_TOPIC",    statistics::PUBLICATION_THROUGHPUT_TOPIC,   num_samples},
+        {"HEARTBEAT_COUNT_TOPIC",           statistics::HEARTBEAT_COUNT_TOPIC,          num_samples},
+        // {"RTPS_SENT_TOPIC",                 statistics::RTPS_SENT_TOPIC,                num_samples},
+        // {"DATA_COUNT_TOPIC",                statistics::DATA_COUNT_TOPIC,               num_samples},
```
Comment (MiguelCompany) [inline test/blackbox/common/DDSBlackboxTestsStatistics.cpp:173]: These lines were already included several lines above and can be removed

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

File: src/cpp/rtps/reader/StatelessReader.cpp (hunk 1 above)
Comment (jparisu) [reply src/cpp/rtps/reader/StatelessReader.cpp]: :+1:

File: src/cpp/rtps/reader/StatelessReader.cpp (hunk 2 above)
Comment (jparisu) [reply src/cpp/rtps/reader/StatelessReader.cpp]: :+1:

File: test/blackbox/common/DDSBlackboxTestsStatistics.cpp (hunk 3 above)
Comment (jparisu) [reply test/blackbox/common/DDSBlackboxTestsStatistics.cpp]: :+1:

File: src/cpp/rtps/reader/StatelessReader.cpp [hunk 4]
```diff
@@ -238,12 +238,14 @@ bool StatelessReader::matched_writer_is_matched(
 bool StatelessReader::change_received(
         CacheChange_t* change)
 {
-    // Only make visible the change if there is not other with bigger sequence number.
+    // Only make visible the change if there is change_received other with bigger sequence number.
```
Comment (MiguelCompany) [inline src/cpp/rtps/reader/StatelessReader.cpp:241]:
```suggestion
    // Only make the change visible if there is not another with a bigger sequence number.
```

**Review by MiguelCompany — APPROVED** [review]
Just one minor nit.

**Review by MiguelCompany — APPROVED** [review]
LGTM

**Pull request merged**